rogue/src/particle_system.rs

```diff
@@ -137,7 +137,7 @@ pub fn update_particle_lifetimes(ecs: &mut World, ctx: &Rltk) {
     {
         let mut particles = ecs.write_storage::<GameAnimationParticle>();
         let entities = ecs.entities();
-        for (entity, mut particle) in (&entities, &mut particles).join() {
+        for (entity, particle) in (&entities, &mut particles).join() {
```

Comment:
There's a few of these where I had, in this case `mut particle`.  I don't really get why this does not need to be marked mutable, we *clearly mutate it* a few lines down. It must be some ECS shenanigans?

Reply:
Ahh, it's not that you aren't mutating `particle`, it's that you've _already_ mutably borrowed it.  As written in the PR, `rust-analyzer` reports the type of particle as `particle: &mut GameAnimationParticle`. I think the warning is just indicating that it's redundant? Changing the line to

```rs
for (entity, particle) in (&entities, &mut particles).join()
```

causes the compiler to correctly complain on the subsequent lines saying

> cannot assign to `particle.lifetime` which is behind a `&` reference, so the data it refers to cannot be written

which I think is consistent with my understanding.